src/jobs/requireTransactions.ts

```diff
@@ -38,18 +40,24 @@ export async function jobRequireTransactions() {
             return;
         }
 
-        const event = findEvent('WithdrawPollCreated', events);
-        if (event) {
-            // TODO Store the events in this receipt to query later
-            // TODO No longer update withdrawals but respond with data based on events in the database
-            await Withdrawal.updateOne(
-                { transactions: String(tx._id) },
-                {
-                    withdrawalId: Number(event.args.id),
-                    poolAddress: assetPool.address,
-                    failReason: '',
-                },
-            );
+        const eventWithdrawPollCreated = findEvent('WithdrawPollCreated', events);
+        if (eventWithdrawPollCreated) {
+            await withdrawal.updateOne({
+                withdrawalId: Number(eventWithdrawPollCreated.args.id),
+                poolAddress: assetPool.address,
+                failReason: '',
+            });
+        }
+
+        const eventRoleGranted = findEvent('RoleGranted', events);
```

Comment:
This seems overly complicated to me... Can't we just assume the user is added as a member and add verify that's registered in the db on every call?

Alternatively this should be taken care of by the event indexer I guess :)